### PR TITLE
feat: allow the GET recurring tasks endpoint to be filtered

### PR DESCRIPTION
**Pull request by mrcpvn**
Endpoint can be filtered by search, page, pageSize, and sort options.

### CODE REVIEW

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsAdminRecurringTaskMonitor-CodeBuildProject
* Commit ID: 3d8c859854ca2d48e5daf0d5eaf3c080e6b45600
* Result: SUCCEEDED
* [Build Logs](https://lrp0k39ym9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1df277cc-6fab-4da1-a951-b5898192fbb8%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/WWTelematics.Admin.RecurringTaskMonitor.Tests/Application/RecurringTaskManagerTest.cs [hunk 1]
```diff
@@ -65,6 +65,116 @@ public void CanReturnEmptyClientGroupList()
...
+
+            Assert.Equal(400, albResponse.StatusCode);
+        }
+
+        /// <summary>
+        /// Test that the GetRecurringTaskList function returns an ALB response with negative search query parameters.
+        /// Negative test case, Returns 200 response code.
+        /// </summary>
+        [Fact]
+        public void CanReturnEmptyRecurringTaskListWithNegativeSearchQueryParametersData()
```
Comment (davidw-foxtrax) [inline src/WWTelematics.Admin.RecurringTaskMonitor.Tests/Application/RecurringTaskManagerTest.cs]: Shouldn't searching for "Label" return a response with a result?

File: src/WWTelematics.Admin.RecurringTaskMonitor/Validators/RecurringTaskModelFilterQueryValidator.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            this.RuleFor(x => x.Page).GreaterThan(0)
+            .WithMessage("Page value must be numeric and should be greater than 0.");
+
+            this.RuleFor(x => x.PageSize).GreaterThan(0)
+            .WithMessage("Pagesize value must be numeric and should be greater than 0.");
+
+            this.When(m => !string.IsNullOrEmpty(m.Sort), () =>
+            {
+                this.RuleFor(x => x.Sort)
+                .Must(x => Enum.IsDefined(typeof(RecurringTaskFilterProperty), x.ToUpperInvariant())) // this.allowedColumns.Contains(x.ToUpperInvariant()))
```
Comment (davidw-foxtrax) [inline src/WWTelematics.Admin.RecurringTaskMonitor/Validators/RecurringTaskModelFilterQueryValidator.cs]: Can we remove the commented code here?

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsAdminRecurringTaskMonitor-CodeBuildProject
* Commit ID: 50cdfb5b58da823935f16698b14218d1a3f20694
* Result: SUCCEEDED
* [Build Logs](https://lrp0k39ym9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e2933259-c959-4eb2-96c2-1ee718dd2139%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/WWTelematics.Admin.RecurringTaskMonitor.Tests/Application/RecurringTaskManagerTest.cs (hunk 1 above)
Comment (mrcpvn) [reply src/WWTelematics.Admin.RecurringTaskMonitor.Tests/Application/RecurringTaskManagerTest.cs]: correct. I changed the test to check if the manager can return an empty result list.

File: src/WWTelematics.Admin.RecurringTaskMonitor/Validators/RecurringTaskModelFilterQueryValidator.cs (hunk 2 above)
Comment (mrcpvn) [reply src/WWTelematics.Admin.RecurringTaskMonitor/Validators/RecurringTaskModelFilterQueryValidator.cs]: removed

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsAdminRecurringTaskMonitor-CodeBuildProject
* Commit ID: 2e1453fc2a94a6913540a1c757bb06b005c2ca02
* Result: SUCCEEDED
* [Build Logs](https://lrp0k39ym9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3b8f4e0f-021b-4a65-a1c1-f043c2e8073d%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsAdminRecurringTaskMonitor-CodeBuildProject
* Commit ID: fb687613e4031c677123bfdce54c37654074ae91
* Result: SUCCEEDED
* [Build Logs](https://lrp0k39ym9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0eab3c5f-60cc-4102-b91d-c6a7731dceee%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by mrcpvn** [comment]
merged into master

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsAdminRecurringTaskMonitor-CodeBuildProject
* Commit ID: cfe0fdbf2e5fae9d322ea053702f491d0f33faf4
* Result: SUCCEEDED
* [Build Logs](https://lrp0k39ym9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ec6588b6-c53a-48b3-95cc-cd43580c96cc%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request closed without merging**